### PR TITLE
Extend customization

### PR DESCRIPTION
**Pull request by ramlmn**
* Add page specific options (including tab-size)
* Fetch real-name with GitHub api if access token is available (needs review)

@sindresorhus @bfred-it This PR needs some review. WDYT?

### CODE REVIEW

**Comment by sindresorhus** [comment]
From the readme:

> We're happy to receive suggestions and contributions, but be aware this is a highly opinionated project. There's [a single commonly-requested option](https://user-images.githubusercontent.com/1402241/27267240-9d2e18c8-54d9-11e7-8a64-971af9e066f3.png) but we're not interested in adding more as it's a slippery slope into adding one for everything. Users will always disagree with something. That being said, we're open to discussing things.

Probably better to open issues about this in the future as we're highly unlikely to add options unless it's something that is highly contentious.

**Comment by bfred-it** [comment]
1. We're not gonna implement both the API call and the normal fetch.
2. Options need to be discussed beforehand.

**Comment by ramlmn** [comment]
Works for me 😊. You may close this PR if its so.

File: src/libs/show-names.js [hunk 1]
```diff
@@ -27,15 +46,17 @@ const fetchName = async username => {
 
 export default async () => {
 	const myUsername = getUsername();
-	const cache = (await getCachedUsers())[storageKey];
+	const cache = (await getCachedUsers())[storageKey] || {};
```
Comment (bfred-it) [inline src/libs/show-names.js]: Please send this line as a separate PR so we can get that in ASAP :)

File: src/content.js [hunk 2]
```diff
@@ -533,33 +537,40 @@ async function onDomReady() {
 
 	const username = getUsername();
 
-	markUnread.setup();
+	// Things that we want to do only when the user is logged in
```
Comment (bfred-it) [inline src/content.js]: I like this. Separate PR? 😃

File: src/libs/show-names.js (hunk 1 above)
Comment (ramlmn) [inline src/libs/show-names.js]: #591
